var/spack/repos/builtin/packages/rmgdft/package.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+from spack import *
+
+
+class Rmgdft(CMakePackage):
+    """RMG is an Open Source code for electronic structure calculations and
+    modeling of materials and molecules. It is based on density functional
+    theory and uses a real space basis and pseudopotentials.
+    """
+
+    homepage = "http://www.rmgdft.org/"
+    url      = "https://github.com/RMGDFT/rmgdft/archive/v4.0.0-beta.3.tar.gz"
+
+    # version('1.2.3', '0123456789abcdef0123456789abcdef')
+    version('4.0.0-beta.3', 'b827762e2da539bf2d41ec5512a7d900')
+
+#    Have not gotten this to work correctly yet.
+#    variant('rmg-cuda', default=False,
+#    description='Base version of the code using cuda')
+
+    # openmpi, mpich etc
+    depends_on('mpi')
+
+    # 1.61 is not the most recent release but newer versions seem to have some
+    # issues with cmake.
+    depends_on('boost@1.61.0%gcc +shared')
```

Comment:
```suggestion
    depends_on('boost@:1.61.0 +shared')
```
Forcing users to use GCC and a specific version of boost is unideal. Boost should compile with any compiler, and mixing compilers in the DAG is never a good idea.

Reply:
Yes the gcc dependence is not necessary but we ran into an issue with boost and cmake. Specifically the cmake module for boost has version specific checks and we ran into cases where the build would fail if the boost version was newer than the cmake version. I know this is not an ideal way of handling this but I just started working with spack and am looking for a better solution.